pmda/simple.py

```diff
@@ -0,0 +1,126 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# PMDA
+# Copyright (c) 2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+"""
+Parallel Analysis helper --- :mod:`pmda.simple`
+===============================================
+
+Collection of decorators to analyze
+
+"""
+from __future__ import absolute_import
+
+from MDAnalysis.core.groups import AtomGroup
+from MDAnalysis.core.universe import Universe
+from MDAnalysis.coordinates.base import ProtoReader
+import numpy as np
+
+from .parallel import ParallelAnalysisBase
+
+
+class AnalysisFromFunction(ParallelAnalysisBase):
+    """
+    Create an analysis from a function working on AtomGroups
+
+    Attributes
+    ----------
+    results : ndarray
+        results of calculation are stored after call to ``run``
+
+    Example
+    -------
+    >>> def rotation_matrix(mobile, ref):
+    >>>     return mda.analysis.align.rotation_matrix(mobile, ref)[0]
+
+    >>> rot = AnalysisFromFunction(rotation_matrix,
+                                   trajectory, mobile, ref).run()
+    >>> print(rot.results)
+
+    Raises
+    ------
+    ValueError : if ``function`` has the same kwargs as ``BaseAnalysis``
+    """
+
+    def __init__(self, function, universe, *args, **kwargs):
+        """
+        Parameters
+        ----------
+        function : callable
+            function to evaluate at each frame
+        Universe : :class:`~MDAnalysis.core.groups.Universe`
+            a :class:`MDAnalysis.core.groups.Universe` (the
+            `atomgroups` must belong to this Universe)
+        *args : list
+           arguments for ``function``
+        **kwargs : dict
+           keyword arguments for ``function``. keyword arguments with name
+           'universe' or 'atomgroups' will be ignored! Atomgroups to analyze
+           can not be passed as keyword arguments currently.
+        """
+
+        self.function = function
+
+        # collect all atomgroups with the same trajectory object as universe
+        trajectory = universe.trajectory
+        arg_ags = []
+        self.other_args = []
+        for arg in args:
+            if isinstance(arg,
+                          AtomGroup) and arg.universe.trajectory == trajectory:
```

Comment:
Only works with `AtomGroup` and not with `ResidueGroup` or `SegmentGroup`?

(Fine with me, can always be changed later, but state assumptions/limitations)